pkg/controller/plan/adapter/ova/kubevirtvmparser.go

```diff
@@ -0,0 +1,56 @@
+package ova
+
+import (
+	"gopkg.in/yaml.v2"
+)
+
+type VirtualMachineInstance struct {
+	ApiVersion string   `yaml:"apiVersion"`
+	Kind       string   `yaml:"kind"`
+	Metadata   Metadata `yaml:"metadata"`
+	Spec       Spec     `yaml:"spec"`
+}
+
+type Metadata struct {
+	Name string `yaml:"name"`
+}
+
+type Spec struct {
+	Domain Domain `yaml:"domain"`
+}
+
+type Domain struct {
+	Firmware Firmware `yaml:"firmware,omitempty"`
+}
+
+type Firmware struct {
+	Bootloader Bootloader `yaml:"bootloader,omitempty"`
+}
+
+type Bootloader struct {
+	Bios *Bios `yaml:"bios,omitempty"`
+	EFI  *EFI  `yaml:"efi,omitempty"`
+}
+
+type Bios struct{}
+
+type EFI struct {
+	SecureBoot bool `yaml:"secureBoot"`
+}
+
+func ReadConfFromYaml(yamlData []byte) (firmware string, err error) {
```

Comment:
the name of this function is not that great since it returns firmware and not vm-configuration